pkg/tsdbctl/check.go

```diff
@@ -91,58 +126,172 @@ func (cc *checkCommandeer) check() error {
 	}
 
 	// Get metric data and metadata
-	allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
 	container, tablePath := cc.rootCommandeer.adapter.GetContainer()
-	objPath := fmt.Sprintf("%s/%s", tablePath, cc.objPath)
+
+	var respChans []chan *v3io.Response
+	if cc.metricName == "" {
+		respChan, err := cc.checkByPath(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using object path '%s'", cc.objPath)
+		}
+		respChans = []chan *v3io.Response{respChan}
+		objPath := "/" + path.Join("", tablePath, cc.objPath)
+		allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
+		input := v3io.GetItemInput{Path: objPath, AttributeNames: allAttrs}
+		_, err = container.GetItem(&input, nil, respChan)
+		if err != nil {
+			return errors.Wrap(err, "failed to send GetItem request")
+		}
+	} else {
+		respChans, err = cc.checkByName(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using metric name (%s) and labels (%v)", cc.metricName, cc.labels)
+		}
+	}
+
+	for _, respChan := range respChans {
+		resp := <-respChan
+		if resp.Error != nil {
+			return errors.Wrap(resp.Error, "GetItem returned with an error")
+		}
+		err = cc.printResponse(resp)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cc *checkCommandeer) checkByPath(container *v3io.Container, tablePath string) (chan *v3io.Response, error) {
+	var err error
+	respChan := make(chan *v3io.Response, 1)
+	objPath := "/" + path.Join("", tablePath, cc.objPath)
+	allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
 	input := v3io.GetItemInput{Path: objPath, AttributeNames: allAttrs}
-	resp, err := container.Sync.GetItem(&input)
+	_, err = container.GetItem(&input, nil, respChan)
 	if err != nil {
-		return errors.Wrap(err, "GetItem failed.")
+		return nil, errors.Wrap(err, "failed to send GetItem request")
+	}
+	return respChan, nil
+}
+
+func (cc *checkCommandeer) checkByName(container *v3io.Container, tablePath string) ([]chan *v3io.Response, error) {
+
+	var err error
+
+	schema, err := getSchema(cc.rootCommandeer.v3iocfg, container)
+	if err != nil {
+		return nil, err
```

Comment:
wrap?

Reply:
I don't think there's a point adding another layer here, just because I broke out a method for readability.

Reply:
This is how we did it almost everywhere...